### PR TITLE
Add `source` fields to error variants for invalid utf-8

### LINKED ISSUE #43: Include source error for InvalidUtf8ToStdout & InvalidUtf8ToStderr

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #43.

### CODE REVIEW

File: src/error.rs [hunk 1]
```diff
@@ -60,10 +62,34 @@ impl std::error::Error for Error {
     fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
         match self {
             Error::CommandIoError { source, .. } => Some(&**source),
-            Error::NoArgumentsGiven
-            | Error::NonZeroExitCode { .. }
-            | Error::InvalidUtf8ToStdout { .. }
-            | Error::InvalidUtf8ToStderr { .. } => None,
+            Error::InvalidUtf8ToStdout { source, .. }
+            | Error::InvalidUtf8ToStderr { source, .. } => Some(&**source),
```
Comment (casey) [inline src/error.rs:66]: Is it not possible to produce a `&dyn` reference without an arc?

**Comment by casey** [comment]
I personally don't like the untyped `source` approach, since I think it's useful to have access to the concrete error type, and useful documentation to see what can cause an error. However, aside from that, this looks good!

File: src/error.rs (hunk 1 above)
Comment (soenkehahn) [reply src/error.rs:66]: Good catch. I thought I needed the `Arc` because `cradle::Error` implements `Clone`. But `FromUtf8Error` does too.

Comment (soenkehahn) [reply src/error.rs]: Never mind. On older rust versions `FromUtf8Error` does *not* implement `Clone`. So I'll put the `Arc` back in.

**Pull request merged**